### PR TITLE
Trace the ProcessRunner

### PR DESCRIPTION
**Pull request by christothes**
To better troubleshoot process based credentials like AzureCli, VS, and VSCode.

related to #20210

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/ProcessRunner.cs [hunk 1]
```diff
@@ -25,6 +25,7 @@ internal sealed class ProcessRunner : IDisposable
 
         public ProcessRunner(IProcess process, TimeSpan timeout, CancellationToken cancellationToken)
         {
+            AzureIdentityEventSource.Singleton.ProcessRunnerInformational($"Running process `{process.StartInfo.FileName}' with arguments {string.Join(", ", process.StartInfo.Arguments)}");
```
Comment (schaabs) [inline sdk/identity/Azure.Identity/src/ProcessRunner.cs:28]: I believe the arguments will contains a resource string in most cases, as well as tenant id if it was specified by the user. We've talked before about whether these are considered PII in the past, but I don't recall the outcome. @g2vinay are we ok to log resource strings and tenant ids or should they be redacted?

Comment (g2vinay) [reply sdk/identity/Azure.Identity/src/ProcessRunner.cs:28]: I believe last we decided these were not PII.
In Java scopes are logged in other credentials when a token is retrieved.

Comment (catalinaperalta) [reply sdk/identity/Azure.Identity/src/ProcessRunner.cs:28]: If I recall correctly, tenantID was not PII but resource strings do get close to being PII depending on the resource name. I'll try to find my notes about this.

Comment (catalinaperalta) [reply sdk/identity/Azure.Identity/src/ProcessRunner.cs:28]: Just checked tenantID is OII (organization identifiable data) so not PII, but I think resources (and therefore their resource strings?) do fall under customer content. I can share the link I have for the data governance site on our teams channel.

Comment (christothes) [reply sdk/identity/Azure.Identity/src/ProcessRunner.cs:28]: Maybe, just to be extra cautious, this change should be coupled with the upcoming setting to log PII information (previously only relevant for MSAL logging).

**Comment by christothes** [comment]
You're welcome, msftbot.

**Comment by christothes** [comment]
dependent on #22852

**Review by schaabs — APPROVED** [review]

**Pull request merged**